src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +30,26 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        
+        splitAddress(address);      
```

Comment:
Need to validate if the input address follows the new format. You can either change the regex or check for 4 comma separated strings.